api/METRICS.md

```diff
@@ -35,72 +45,260 @@ acc_ratio = accuracy_score_ratio(y_true, y_pred, sensitive_features=sf, **other_
+#
+# Alternatively, `metric` can be a metric group summary function, and `transformation`
+# can be a function one of the functions <transformation>_from_summary.
+
+make_derived_metric(metric, transformation) [TODO]
+# return a callable object <metric>_<transformation>:
+# <metric>_<transformation>(...) = derived_metric(metric, transformation, ...)
+
+# Predefined metrics are named according to the following patterns:
 <metric>_group_summary(y_true, y_pred, *, sensitive_features, **other_kwargs)
 <metric>_difference(y_true, y_pred, *, sensitive_features, **other_kwargs)
 <metric>_ratio(y_true, y_pred, *, sensitive_features, **other_kwargs)
 <metric>_group_min(y_true, y_pred, *, sensitive_features, **other_kwargs)
 <metric>_group_max(y_true, y_pred, *, sensitive_features, **other_kwargs)
 ```
 
-*Transformations and transformation codes*
+### Definitions of transformation functions
 
-|transformation function|output|metric-specific function|code|aif360|
+|transformation function|output|derived metric name|code|aif360|
 |-----------------------|------|------------------------|----|------|
 |`difference_from_summary`|max - min|`<metric>_difference`|D|unprivileged - privileged|
 |`ratio_from_summary`|min / max|`<metric>_ratio`|R| unprivileged / privileged|
 |`group_min_from_summary`|min|`<metric>_group_min`|Min| N/A |
 |`group_max_from_summary`|max|`<metric>_group_max`|Max| N/A |
 
-*Tasks and task codes*
+### List of predefined metrics
 
-|task|definition|code|
-|----|----------|----|
-|binary classification|labels and predictions are in {0,1}|class|
-|probabilistic binary classification|labels are in {0,1}, predictions are in [0,1] and correspond to estimates of P(y\|x)|prob|
-|randomized binary classification|labels are in {0,1}, predictions are in [0,1] and represent the probability of drawing y=1 in a randomized decision|class-rand|
-|regression|labels and predictions are real-valued|reg|
+* In the list of predefined metrics, we refer to the following machine learning tasks:
 
-*Predefined metric-specific functions*
+  |task|definition|code|
+  |----|----------|----|
+  |binary classification|labels and predictions are in {0,1}|class|
+  |probabilistic binary classification|labels are in {0,1}, predictions are in [0,1] and correspond to estimates of P(y\|x)|prob|
+  |randomized binary classification|labels are in {0,1}, predictions are in [0,1] and represent the probability of drawing y=1 in a randomized decision|class-rand|
+  |regression|labels and predictions are real-valued|reg|
+
+* For each _base metric_, we provide the list of predefined derived metrics, using D, R, Min, Max to refer to the transformations from the table above, and G to refer to `<metric>_group_summary`. We follow these rules:
+  * always provide G (except for demographic parity and equalized odds, which do not make sense as group-level metrics)
+  * provide D and R for confusion-matrix metrics
+  * provide Min for score functions (worst-case score)

```

Comment:
I can't think of any such cases, but are there scores people could come up with that would work the other way round? Or would you just consider them losses then?